Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:14
+
+LABEL maintainer="Ryan Blunden <ryan.blunden@ext.doppler.com>"
+
+# DOPPLER_TOKEN must be supplied as a build-arg
+ARG DOPPLER_TOKEN
+ENV DOPPLER_TOKEN ${DOPPLER_TOKEN}
+
+# Install the Doppler CLI
+RUN curl -Ls https://cli.doppler.com/install.sh | sh
+
+# Execute `doppler run` so it creates a secrets fallback file
+RUN doppler run -- echo "Create secrets fallback file"
+
+WORKDIR /usr/src/app
+COPY ["package.json", "package-lock.json*", "src", "./"]
+RUN npm install --production --silent && mv node_modules ../
+COPY . .
+
+CMD ["doppler", "run", "--", "npm", "start"]
```

Comment:
As a last step before execution we can switch to the pre-existing `node` user, to ensure our app doesn't run with root perms. You can see an example of this in our server's Dockerfile [here](https://github.com/DopplerHQ/server/blob/develop/deployments/Dockerfile#L40).